### ISSUE TITLE
panic: runtime error: slice bounds out of range  in  decodeUTF16

### ISSUE THREAD

**Issue by GithubUser5462** [dhowden/tag#34]
Hello, i stumbled upon a slice bounds out of range bug.
I'm not sure what is the cause but it happens every time i try to process more than 79 files.
From my testing it doesn't seem to be related to a specific file.

Crash log:
```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/dhowden/tag.decodeUTF16(0xc042075053, 0x15, 0x15, 0x545180, 0x5f2908, 0xc042075051, 0x20)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2frames.go:406 +0x1d4
github.com/dhowden/tag.decodeUTF16WithBOM(0xc042075051, 0x17, 0x17, 0xc04206ac90, 0x2)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2frames.go:400 +0x9f
github.com/dhowden/tag.decodeText(0xc042075001, 0xc042075051, 0x17, 0x17, 0xc04206ac90, 0x2, 0x2, 0x0)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2frames.go:329 +0x82
github.com/dhowden/tag.readTextWithDescrFrame(0xc042075050, 0x21, 0x21, 0x100, 0x5f2f60, 0x0, 0x0)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2frames.go:460 +0x1bb
github.com/dhowden/tag.readID3v2Frames(0x544880, 0xc042076070, 0x120, 0xc042050d60, 0xc042076070, 0x0, 0x0)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2.go:305 +0xddf
github.com/dhowden/tag.ReadID3v2Tags(0x544ae0, 0xc042076070, 0x1, 0x0, 0x0, 0x0)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/id3v2.go:405 +0xe7
github.com/dhowden/tag.ReadFrom(0x544ae0, 0xc042076070, 0x5450c0, 0xc0422c5a40, 0x0, 0x0)
	D:/Projects/go-projects/DirScan/go/src/github.com/dhowden/tag/tag.go:53 +0x2a0
main.getArtistAlbumTitleYear(0xc0420a1030, 0x61, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	D:/Projects/go-projects/DirScan/go/src/main/main.go:259 +0xea
main.main.func1(0xc0420a1030, 0x61, 0x5450c0, 0xc0422c59e0, 0x0, 0x0, 0x0, 0x0)
	D:/Projects/go-projects/DirScan/go/src/main/main.go:77 +0x358
path/filepath.walk(0xc0420a1030, 0x61, 0x5450c0, 0xc0422c59e0, 0xc042050480, 0x0, 0x0)
	C:/Go/src/path/filepath/path.go:357 +0x409
path/filepath.walk(0xc04207c240, 0x3f, 0x5450c0, 0xc0422c5740, 0xc042050480, 0x0, 0x0)
	C:/Go/src/path/filepath/path.go:381 +0x2c9
path/filepath.walk(0x5333bf, 0x25, 0x5450c0, 0xc042098060, 0xc042050480, 0x0, 0x20)
	C:/Go/src/path/filepath/path.go:381 +0x2c9
path/filepath.Walk(0x5333bf, 0x25, 0xc042050480, 0x12, 0x0)
	C:/Go/src/path/filepath/path.go:403 +0x10d
main.main()
	D:/Projects/go-projects/DirScan/go/src/main/main.go:53 +0x481
```

**Comment by dhowden** [comment]
Hi!  Thanks for the report!

It looks like the files you are trying to read have invalid UTF-16 encoded text in them.  I've just pushed a fix ba105bf to cause this to error instead of throwing a panic, so let me know how it goes.  This should at least stop your program from crashing.

Closing this issue for now, feel free to re-open if you are still getting a crash with the fix applied (or open a new issue).

**Comment by GithubUser5462** [comment]
Ah yes, you were right, the tags are definitively at fault.
Well at least now it's easier to handle the error. Thank you.
